lib/src/executables/sort_intl.dart

```diff
@@ -0,0 +1,156 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:args/args.dart';
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:logging/logging.dart';
+import 'package:over_react_codemod/src/intl_suggestors/intl_messages.dart';
+
+import 'package:over_react_codemod/src/util/package_util.dart';
+import 'package:path/path.dart' as p;
+import '../util.dart';
+import 'intl_message_migration.dart';
+
+final _log = Logger('orcm.intl_message_migration');
+
+const _noMigrate = 'no-migrate';
+
+final FileSystem fs = const LocalFileSystem();
+
+final parser = ArgParser()
+  ..addFlag(
+    'help',
+    abbr: 'h',
+    negatable: false,
+    help: 'Prints this help output.',
+  )
+  ..addFlag(_noMigrate,
+      negatable: false,
+      defaultsTo: false,
+      help:
+      'Does not run any migrators, overriding any --migrate flags. Can still be used with --prune-unused, and '
+          'will force the messages file to be sorted and rewritten'
+  );
+
+late ArgResults parsedArgs;
+
+void main(List<String> args) async {
+  parsedArgs = parser.parse(args);
+  if (parsedArgs['help'] as bool) {
+    printUsage();
+    return;
+  }
+
+  // If we have specified paths on the command line, limit our processing to
+  // those, and make sure they're absolute.
+  var basicDartPaths = parsedArgs.rest.isEmpty ? ['lib'] : parsedArgs.rest;
+  var dartPaths = [
+    for (var path in basicDartPaths) p.canonicalize(p.absolute(path))
+  ];
+
+  // Work around parts being unresolved if you resolve them before their libraries.
+  // TODO - reference analyzer issue for this once it's created
+  final packageRoots = dartPaths.map(findPackageRootFor).toSet().toList();
+  packageRoots.sort((packageA, packageB) =>
+  p.split(packageB).length - p.split(packageA).length);
+
+  // TODO: Use packageConfig and utilities for reading that rather than manually parsing pubspec..
+  Map<String, String> packageNameLookup = {
+    for (String path in pubspecYamlPaths())
+      p.basename(findPackageRootFor(path)): fs
+          .file(path)
+          .readAsLinesSync()
+          .firstWhere((line) => line.startsWith('name'))
+          .split(':')
+          .last
+          .trim()
+  };
+
+  final processedPackages = Set<String>();
+  await pubGetForAllPackageRoots(dartPaths);
+
+  for (String package in packageRoots) {
+    final packageRoot = p.basename(package);
+    final packageName = packageNameLookup[packageRoot] ?? 'fix_me_bad_name';
+    _log.info('Starting migration for $packageName');
+    List<String> packageDartPaths;
+    try {
+      packageDartPaths =
+          dartFilesToMigrateForPackage(package, processedPackages).toList();
+    } on FileSystemException {
+      _log.info('${package} does not have a lib directory, moving on...');
+      return;
+    }
+
+    packageDartPaths = limitPaths(packageDartPaths, allowed: dartPaths);
+    sortPartsLast(packageDartPaths);
+
+    final IntlMessages messages = IntlMessages(packageName);
```

Comment:
It's possible to create an IntlMessage and give it an outputFileName parameter. I strongly suspect that if you just found all the _intl.dart files underneath the directory and looped over them giving them the main package name (or possibly even an empty string) but also the outputFilename that that would be enough.